pyodide_build/pypabuild.py

```diff
@@ -284,17 +284,23 @@ def make_command_wrapper_symlinks(symlink_dir: Path) -> dict[str, str]:
     return env
 
 
-# TODO: a context manager is no longer needed here
-@contextmanager
-def _create_symlink_dir(
-    env: dict[str, str],
-    build_dir: Path,
-):
-    # Leave the symlinks in the build directory. This helps with reproducing.
-    symlink_dir = build_dir / "pywasmcross_symlinks"
-    shutil.rmtree(symlink_dir, ignore_errors=True)
-    symlink_dir.mkdir()
-    yield symlink_dir
+def _create_symlink_dir(build_dir: Path | None) -> Path:
+    if build_dir is None:
+        # If no build directory is provided, create a temporary directory
+
+        # NOTE: This is not ideal and should not be merged as-is. We need
```

Comment:
Is there anything that wrong with this? Temporary directories get removed on reboot anyways. Also wouldn't the alternative be... using a context manager?